libcloud/compute/drivers/gce.py

```
@@ -1211,9 +1211,16 @@ def create_node(self, name, size, image, location=None,
                                               use_existing=use_existing_disk,
                                               ex_disk_type=ex_disk_type)
 
-        if ex_metadata is not None:
-            ex_metadata = {"items": [{"key": k, "value": v}
-                                     for k, v in ex_metadata.items()]}
+        if not ex_metadata or not isinstance(ex_metadata, dict):
+            # Should output a relevant error message if non-dict is received?
+            ex_metadata = None
```

We should throw `ValueError` if user passes in an invalid value (not a dictionary) instead of silently ignoring it.